pandas/core/indexes/multi.py

```diff
@@ -1359,6 +1359,9 @@ def is_monotonic_increasing(self):
         increasing) values.
         """
 
+        if all([x.is_monotonic for x in self.levels]):
+            return libalgos.is_lexsorted([x.astype("int64") for x in self.codes])
```

Comment:
you maybe be able to add copy=False here

Reply:
Looks like that nets another ~10-20% gain - thanks!

Reply:
lol